src/main/java/io/cryostat/net/web/http/api/v2/RecordingMetadataPatchHandler.java

```diff
@@ -0,0 +1,137 @@
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.v2;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import javax.inject.Inject;
+
+import io.cryostat.messaging.notifications.NotificationFactory;
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.AbstractAuthenticatedRequestHandler;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.recordings.RecordingArchiveHelper;
+import io.cryostat.recordings.RecordingMetadataManager;
+import io.cryostat.recordings.RecordingNotFoundException;
+
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+
+public class RecordingMetadataPatchHandler extends AbstractAuthenticatedRequestHandler {
+
+    static final String PATH = "recordings/:recordingName/labels";
+
+    private final RecordingArchiveHelper recordingArchiveHelper;
+    private final RecordingMetadataManager recordingMetadataManager;
+    private final NotificationFactory notificationFactory;
+
+    @Inject
+    RecordingMetadataPatchHandler(
+            AuthManager auth,
+            RecordingArchiveHelper recordingArchiveHelper,
+            RecordingMetadataManager recordingMetadataManager,
+            NotificationFactory notificationFactory) {
+        super(auth);
+        this.recordingArchiveHelper = recordingArchiveHelper;
+        this.recordingMetadataManager = recordingMetadataManager;
+        this.notificationFactory = notificationFactory;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2_1;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.PATCH;
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return Set.of(ResourceAction.READ_TARGET, ResourceAction.READ_RECORDING);
```

Comment:
Since this handler operates on archived recordings, I don't think it needs the `READ_TARGET` permission. I think `UPDATE_RECORDING` might make sense, if we consider the metadata/labels of a recording to be a part of the recording resource itself.